common/test/acceptance/tests/studio/test_studio_html_editor.py

```diff
@@ -100,3 +113,186 @@ def test_link_plugin_sets_url_correctly(self):
+                Then the page text contains:
+                ""
+                    <math><msup><mi>x</mi><mn>2</mn></msup></math>
+                ""
+        """
+        content = "<math><msup><mi>x</mi><mn>2</mn></msup></math>"
+        self._content_test(content)
+
+    def test_code_format_toolbar_wraps_text_with_code_tags(self):
+        """
+        Scenario: Code format toolbar button wraps text with code tags
+        Given I have created a Blank HTML Page
+            When I edit the page
+            And I set the text to "display as code" and I select the text
+            And I save the page
+                Then the page text contains:
+                ""
+                    <p><code>display as code</code></p>
+                ""
+        """
+        self.html_editor.set_text_and_select("display as code")
+        self.html_editor.click_code_toolbar_button()
+        self.html_editor.save_content()
+        html = self.container_page.content_html
+        self.assertIn('<p><code>display as code</code></p>', html)
+
+    def test_raw_hmlcomponent_does_not_change_text(self):
+        """
+        Scenario: Raw HTML component does not change text
+        Given I have created a raw HTML component
+            When I edit the page
+            And type "<li>zzzz<ol> " into the Raw Editor
+            And I save the page
+                Then the page text contains:
+                  ""
+                  <li>zzzz<ol>
+                  ""
+            And I edit the page
+                Then the Raw Editor contains exactly:
+                  ""
+                  <li>zzzz<ol>
+                  ""
+        """
+        content = "<li>zzzz<ol>"
+        self.html_editor.set_content_in_tinymce_editor(content)
+        self.html_editor.save_content()
+        html = self.container_page.content_html
+        self.assertIn(content, html)
+        self.container_page.edit()
+        editor_value = self.html_editor.editor_value
+        self.assertEqual(content, editor_value)
+
+    def test_tinymce_toolbar_buttons_are_as_expected(self):
+        """
+        Scenario: TinyMCE toolbar buttons are as expected
+        Given I have created a Blank HTML Page
+        When I edit the page
+            Then the expected toolbar buttons are displayed
+        """
+        expected_buttons = [
+            u'bold',
+            u'italic',
+            u'underline',
+            u'forecolor',
+            # This is our custom "code style" button, which uses an image instead of a class.
+            u'none',
+            u'alignleft',
+            u'aligncenter',
+            u'alignright',
+            u'alignjustify',
+            u'bullist',
+            u'numlist',
+            u'outdent',
+            u'indent',
+            u'blockquote',
+            u'link',
+            u'unlink',
+            u'image'
+        ]
+        toolbar_dropdowns = self.html_editor.toolbar_dropdown_titles
+        self.assertEqual(len(toolbar_dropdowns), 2)
```

Comment:
you can add a one liner comments on top of assertions which are not self explanatory